src/expr/src/relation/func.rs

```diff
@@ -1898,7 +1898,7 @@ impl RustType<ProtoAnalyzedRegex> for AnalyzedRegex {
 
 impl AnalyzedRegex {
     pub fn new(s: &str) -> Result<Self, regex::Error> {
```

Comment:
```suggestion
    pub fn new(s: String) -> Result<Self, regex::Error> {
```

Using a String might avoid an allocation if the caller already has a String.

Reply:
Thx, changed